Source/Core/Duality/Utility/Coroutines/Coroutine.cs

```diff
@@ -0,0 +1,125 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Duality.Resources;
+
+namespace Duality
+{
+	public sealed class Coroutine : IDisposable
+	{
+		/// <summary>
+		/// Starts a coroutine and registers it in the scene's CoroutineManager
+		/// </summary>
+		/// <param name="scene">The scene that will execute the coroutine</param>
+		/// <param name="method">The coroutine implementation</param>
+		/// <param name="name">The name of the coroutine, optional</param>
+		/// <returns>The created corutine</returns>
+		public static Coroutine Start(Scene scene, IEnumerable<IWaitCondition> method, string name = null)
+		{
+			return Start(scene, method.GetEnumerator(), name);
+		}
+
+		/// <summary>
+		/// Starts a coroutine and registers it in the scene's CoroutineManager
+		/// </summary>
+		/// <param name="scene">The scene that will execute the coroutine</param>
+		/// <param name="method">The coroutine implementation</param>
+		/// <param name="name">The name of the coroutine, optional</param>
+		/// <returns>The created corutine</returns>
+		public static Coroutine Start(Scene scene, IEnumerator<IWaitCondition> enumerator, string name = null)
+		{
+			return scene.CoroutineManager.StartNew(enumerator, name);
+		}
+
+		private IEnumerator<IWaitCondition> enumerator = null;
+
+		/// <summary>
+		/// Returns true if there are no more CoroutineActions to be performed
+		/// </summary>
+		public CoroutineStatus Status { get; private set; }
```

Comment:
Comment seems incorrect. I'm guessing this used to be a boolean property.

Reply:
Fixed